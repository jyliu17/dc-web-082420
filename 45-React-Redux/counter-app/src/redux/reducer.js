const initialState = {
   count: 0
}
 
const reducer = (oldState=initialState, action) => {
 
 switch(action.type){
   case "INCREMENT":
     return {count: oldState.count + action.payload}
   case "DECREMENT":
     return {count: oldState.count - action.payload}
   default:
     return oldState
 }
}
 
export default reducer