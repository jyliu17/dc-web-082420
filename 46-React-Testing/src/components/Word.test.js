import React from 'react'
import Word from './Word'
import {createStore} from 'redux'
import manageWords from '../reducers/manageWords'
import renderer from 'react-test-renderer'

let mockStore = createStore(manageWords)
let mockProps = {
    word: {
        id: 1,
        name: "fuglen",
        img_url: null
    }
}

describe('Word component', () => {
    it('renders a word correctly', () => {
        const tree = renderer.create(<Word {...mockProps} store={mockStore}/>).toJSON()
        expect(tree).toMatchSnapshot()
    })
})