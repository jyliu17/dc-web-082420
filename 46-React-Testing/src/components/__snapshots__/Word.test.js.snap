// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Word component renders a word correctly 1`] = `
<div
  className="word-card"
>
  <h2>
    fuglen
  </h2>
  <img
    className="word-image"
    src={null}
  />
  <br />
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</div>
`;
